backend/tests/Designer.Tests/Services/OptionsServiceTests.cs

```diff
@@ -0,0 +1,264 @@
+        const string repo = "app-with-options"; // Has 2 option lists
+        string targetRepository = TestDataHelper.GenerateTestRepoName();
+        await TestDataHelper.CopyRepositoryForTest(Org, repo, Developer, targetRepository);
+
+        // Act
+        var optionsService = GetOptionsServiceForTest();
+        string[] optionListIds = optionsService.GetOptionListIds(Org, targetRepository, Developer);
+
+        // Assert
+        Assert.Equal(2, optionListIds.Length);
+    }
+
+    [Fact]
+    public async Task GetOptionListIds_ShouldReturnEmptyArray_WhenOptionListsDoesNotExist()
+    {
+        // Arrange
+        const string repo = "empty-app"; // Has no option lists
+        string targetRepository = TestDataHelper.GenerateTestRepoName();
+        await TestDataHelper.CopyRepositoryForTest(Org, repo, Developer, targetRepository);
+
+        // Act
+        var optionsService = GetOptionsServiceForTest();
+        string[] optionListIds = optionsService.GetOptionListIds(Org, targetRepository, Developer);
+
+        // Assert
+        Assert.Empty(optionListIds);
+    }
+
+    [Fact]
+    public async Task GetOptions_ShouldReturnOptions_WhenOptionsExists()
+    {
+        // Arrange
+        // This  option list matches the options in 'app-with-options'
+        var expectedOptions = new List<Option>
+        {
+            new Option
+            {
+                Label = "label1",
+                Value = "value1",
+            },
+            new Option
+            {
+                Label = "label2",
+                Value = "value2",
+            }
+        };
+
+        const string repo = "app-with-options";
+        const string optionListId = "test-options";
+        string targetRepository = TestDataHelper.GenerateTestRepoName();
+        await TestDataHelper.CopyRepositoryForTest(Org, repo, Developer, targetRepository);
+
+        // Act
+        var optionsService = GetOptionsServiceForTest();
+        var fetchedOptions = await optionsService.GetOptions(Org, targetRepository, Developer, optionListId);
+
+        // Assert
+        Assert.Equal(expectedOptions.Count, fetchedOptions.Count);
+
+        for (int i = 0; i < expectedOptions.Count; i++)
+        {
+            Assert.Equal(expectedOptions[i].Label, fetchedOptions[i].Label);
+            Assert.Equal(expectedOptions[i].Value, fetchedOptions[i].Value);
+            Assert.Equal(expectedOptions[i].Description, fetchedOptions[i].Description);
+            Assert.Equal(expectedOptions[i].HelpText, fetchedOptions[i].HelpText);
+        }
+    }
+
+    [Fact]
+    public async Task GetOptions_ShouldThrowNotFoundException_WhenOptionsDoesNotExist()
+    {
+        // Arrange
+        const string repo = "empty-app"; // Has no option lists
+        const string optionListId = "test-options";
+
+        string targetRepository = TestDataHelper.GenerateTestRepoName();
+        await TestDataHelper.CopyRepositoryForTest(Org, repo, Developer, targetRepository);
```

Comment:
I believe we do not need to create testRepos when we do get 🧐

Reply:
Nice catch! Fixed ✅